velox/exec/HashBuild.cpp

```diff
@@ -981,4 +982,34 @@ bool HashBuild::testingTriggerSpill() {
       spillConfig()->testSpillPct;
 }
 
+bool HashBuild::canReclaim() const {
+  return Operator::canReclaim() && (state_ == State::kRunning);
+}
+
+void HashBuild::reclaim(uint64_t targetBytes) {
+  VELOX_CHECK(canReclaim());
+  auto* driverCtx = operatorCtx_->driverCtx();
+  VELOX_CHECK(!driverCtx->driver->state().isOnThread());
+  const auto& task = operatorCtx_->task();
+  VELOX_CHECK(task->pauseRequested());
```

Comment:
Should we pause the task manually before calling the method?

Reply:
Yeah, we will call this at the task memory reclaimer which is in followup PR.